### PR TITLE
Fixed #32420 -- Fixed detecting primary key values in deserialization when PK is also a FK.

### PR DESCRIPTION
**Pull request by trybik**
Implements directly info provided here, incl. the test exposing the issue: https://code.djangoproject.com/ticket/32420

### CODE REVIEW

File: tests/serializers/models/base.py [hunk 1]
```diff
@@ -52,6 +52,21 @@ def __str__(self):
...
+        raise NotImplementedError("Don't get by natural key")
+
+
+class GhostWriter(models.Model):
+    author_ptr = models.ForeignKey(Author, on_delete=models.CASCADE, primary_key=True)
+
+    objects = GhostWriterManager()
+
+    def natural_key(self):
+        raise NotImplementedError("There is no natural key")
```
Comment (felixxm) [inline tests/serializers/models/base.py:67]:
```suggestion
        raise NotImplementedError('This method was not expected to be called.')
```

File: tests/serializers/models/base.py [hunk 2]
```diff
@@ -52,6 +52,21 @@ def __str__(self):
         return self.name
 
 
+class GhostWriterManager(models.Manager):
+
+    def get_by_natural_key(self, *args, **kwargs):
+        raise NotImplementedError("Don't get by natural key")
```
Comment (felixxm) [inline tests/serializers/models/base.py:58]: It should be enough to raise `NotImplementedError` in `natural_key()`, so I would return a base implementation here, e.g.
```suggestion
        return self.get(*args, **kwargs)
```

File: tests/serializers/models/base.py [hunk 3]
```diff
@@ -52,6 +52,21 @@ def __str__(self):
...
+
+
+class GhostWriter(models.Model):
```
Comment (felixxm) [inline tests/serializers/models/base.py:61]:
```suggestion
class FKAsPKNoNaturalKey(models.Model):
```

File: tests/serializers/models/base.py [hunk 4]
```diff
@@ -52,6 +52,21 @@ def __str__(self):
         return self.name
 
 
+class GhostWriterManager(models.Manager):
```
Comment (felixxm) [inline tests/serializers/models/base.py:55]:
```suggestion
class FKAsPKNoNaturalKeyManager(models.Manager):
```

File: tests/serializers/test_json.py [hunk 5]
```diff
@@ -92,6 +92,29 @@ def test_json_deserializer_exception(self):
             for obj in serializers.deserialize("json", """[{"pk":1}"""):
                 pass
 
+    def test_different_primary_key_name_and_attname_with_broken_natural_key(self):
```
Comment (felixxm) [inline tests/serializers/test_json.py:95]:
```suggestion
    def test_fk_as_pk_natural_key_not_called(self):
```

File: tests/serializers/test_json.py [hunk 6]
```diff
@@ -92,6 +92,29 @@ def test_json_deserializer_exception(self):
...
+        }, {
+            "pk": "1",
+            "model": "serializers.ghostwriter",
+            "fields": {
+            }
+        }]"""
+        dobjs = list(serializers.deserialize('json', test_string))
+        self.assertEqual(len(dobjs), 2)
+        for dobj in dobjs:
+            self.assertEqual(dobj.object.pk, 1)
```
Comment (felixxm) [inline tests/serializers/test_json.py:116]: This is not a `JSON`-specific test, so can we move it to the `tests/serializers/test_natural.py`?

**Review by felixxm — COMMENTED** [review]
@trybik Thanks for this patch :+1:

File: tests/serializers/test_json.py (hunk 6 above)
Comment (trybik) [reply tests/serializers/test_json.py:116]: Yes, it's not `JSON`-specific - makes perfect sense to move to `test_nautral.py`.

**Comment by felixxm** [comment]
@trybik Please check my comments, and update your patch.

**Comment by trybik** [comment]
> @trybik Please check my comments, and update your patch.

yeap, just done it - please review again

**Comment by trybik** [comment]
@felixxm or pls wait still a sec - fighting w/ linter

**Comment by trybik** [comment]
@felixxm ok, now should be fine - sorry for fuss

**Review by felixxm — COMMENTED** [review]
@trybik Thanks for updates :+1: Welcome aboard :boat: 

I pushed minor edits.

**Pull request merged**